modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryStorageMetrics.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.pagememory;
+
+import org.apache.ignite.internal.lang.IgniteInternalCheckedException;
+import org.apache.ignite.internal.metrics.LongGauge;
+import org.apache.ignite.internal.pagememory.persistence.store.FilePageStore;
+import org.apache.ignite.internal.pagememory.persistence.store.FilePageStoreManager;
+import org.apache.ignite.internal.pagememory.persistence.store.GroupPageStoresMap.GroupPartitionPageStore;
+
+/** Persistent page memory storage metrics. */
+class PersistentPageMemoryStorageMetrics {
+    /** Initializes metrics in the given metric source. */
+    static void initMetrics(
+            PersistentPageMemoryStorageMetricSource source,
+            FilePageStoreManager filePageStoreManager
+    ) {
+        source.addMetric(new LongGauge(
+                "StorageSize",
+                "Size of the entire storage in bytes.",
+                () -> storageSize(filePageStoreManager)
+        ));
+    }
+
+    private static long storageSize(FilePageStoreManager filePageStoreManager) {
+        return filePageStoreManager.allPageStores().mapToLong(PersistentPageMemoryStorageMetrics::fullSize).sum();
+    }
+
+    private static long fullSize(GroupPartitionPageStore<FilePageStore> pageStore) {
+        try {
+            return pageStore.pageStore().fullSize();
+        } catch (IgniteInternalCheckedException e) {
+            return 0;
```

Comment:
This looks suspicious. Does any contract requires us to suppress exceptions? How about just throwing the exception (wrapped in something else)? Otherwise we'll not even know that something went wrong.

Reply:
I didn't find such a contract. I assumed that if something goes wrong only for a metric, it can be ignored because it won't affect the cluster's operation.
I view the metric as a non-critical resource, so I don't have to be so strict.

Reply:
Do you think it's worth it?
Most likely, there will be warnings that won't do anything.